src/mon/MgrMonitor.cc

```diff
@@ -524,6 +524,13 @@ bool MgrMonitor::prepare_beacon(MonOpRequestRef op)
       pending_map.available_modules = m->get_available_modules();
       updated = true;
     }
+    const auto& clients = m->get_clients();
```

Comment:
I think this will only trigger when we are first going active.. but the mgr modules might start up their rados client(s) later.  I think we probably want an explicit comparison to see if the beacon's addrvec list is different than the registered list, and update then?

Reply:
In my testing, this catches all updates.